// prisma-nested-middleware/dist/esm/lib/utils/targets.js
import { isQueryAction, isReadAction, isWriteAction } from "./actions";
export function isQueryTarget(target) {
    return isQueryAction(target.action);
}
export function isReadTarget(target) {
    return isReadAction(target.action);
}
export function isWriteTarget(target) {
    return isWriteAction(target.action);
}
export function buildOperationsPath(operations) {
    if (!operations)
        return [];
    return operations.flatMap((op) => {
        if (typeof op.index === "number")
            return [op.logicalOperator, op.index.toString()];
        return [op.logicalOperator];
    });
}
export function buildQueryTargetPath(target) {
    const path = target.parentTarget ? buildTargetPath(target.parentTarget) : [];
    if (!target.relationName) {
        return [...path, target.action];
    }
    if (target.operations) {
        path.push(...buildOperationsPath(target.operations));
    }
    if (target.readAction) {
        path.push(target.readAction);
    }
    path.push(target.relationName);
    if (target.readAction) {
        path.push("where");
    }
    if (target.modifier) {
        path.push(target.modifier);
    }
    return path;
}
export function buildWriteTargetPath(target) {
    const path = target.parentTarget ? buildTargetPath(target.parentTarget) : [];
    if (target.field) {
        path.push(target.field);
    }
    path.push(target.relationName, target.action);
    if (typeof target.index === "number") {
        path.push(target.index.toString());
    }
    return path;
}
export function buildReadTargetPath(target) {
    const path = target.parentTarget ? buildTargetPath(target.parentTarget) : [];
    if (!target.relationName) {
        return [...path, target.action];
    }
    if (!target.field) {
        return [...path, target.action, target.relationName];
    }
    return [...path, target.field, target.relationName, target.action];
}
export function buildTargetPath(target) {
    if (isQueryTarget(target))
        return buildQueryTargetPath(target);
    if (isReadTarget(target))
        return buildReadTargetPath(target);
    return buildWriteTargetPath(target);
}
export const buildTargetRelationPath = (target) => {
    if (!isReadTarget(target))
        return null;
    if (target.parentTarget) {
        const basePath = buildTargetRelationPath(target.parentTarget);
        if (!basePath)
            return null;
        return target.relationName ? [...basePath, target.relationName] : basePath;
    }
    return target.relationName ? [target.relationName] : [];
};
export function targetChainLength(target, count = 0) {
    if (!target.parentTarget) {
        return count + 1;
    }
    return targetChainLength(target.parentTarget, count + 1);
}
