// prisma-nested-middleware/dist/lib/utils/actions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.logicalOperators = exports.modifiers = exports.isWriteAction = exports.isReadAction = exports.isQueryAction = exports.toOneRelationNonListActions = exports.writeActions = exports.readActions = exports.queryActions = void 0;
exports.queryActions = ["where"];
exports.readActions = ["include", "select"];
exports.writeActions = [
    "create",
    "update",
    "upsert",
    "createMany",
    "updateMany",
    "delete",
    "deleteMany",
    "disconnect",
    "connect",
    "connectOrCreate",
];
exports.toOneRelationNonListActions = [
    "create",
    "update",
    "delete",
    "upsert",
    "connect",
    "connectOrCreate",
    "disconnect",
];
function isQueryAction(action) {
    return exports.queryActions.includes(action);
}
exports.isQueryAction = isQueryAction;
function isReadAction(action) {
    return exports.readActions.includes(action);
}
exports.isReadAction = isReadAction;
function isWriteAction(action) {
    return exports.writeActions.includes(action);
}
exports.isWriteAction = isWriteAction;
exports.modifiers = ["is", "isNot", "some", "none", "every"];
exports.logicalOperators = ["AND", "OR", "NOT"];
